File: src/main/java/me/dmillerw/loreexpansion/proxy/CommonProxy.java
package me.dmillerw.loreexpansion.proxy;

import me.dmillerw.loreexpansion.LoreExpansion;
import net.minecraftforge.fml.common.network.NetworkRegistry;

public class CommonProxy {

    public void preInit() {

    }

    public void init() {
        NetworkRegistry.INSTANCE.registerGuiHandler(LoreExpansion.INSTANCE, new GuiHandler());
    }

    public void postInit() {

    }
}

File: src/main/kotlin/me/dmillerw/loreexpansion/proxy/CommonProxy.kt
package me.dmillerw.loreexpansion.proxy

import me.dmillerw.loreexpansion.LoreExpansion
import me.dmillerw.loreexpansion.core.LoreLoader
import me.dmillerw.loreexpansion.core.player.PlayerEventHandler
import me.dmillerw.loreexpansion.item.ItemLore
import net.minecraftforge.common.MinecraftForge
import net.minecraftforge.fml.common.event.FMLInitializationEvent
import net.minecraftforge.fml.common.event.FMLPostInitializationEvent
import net.minecraftforge.fml.common.event.FMLPreInitializationEvent

/**
 * @author dmillerw
 */
open class CommonProxy {

    public open fun preInit(event: FMLPreInitializationEvent) {
        ItemLore.register()

        LoreLoader.initialize(LoreExpansion.loreFolder)

        MinecraftForge.EVENT_BUS.register(PlayerEventHandler)
    }

    public open fun init(event: FMLInitializationEvent) {

    }

    public open fun postInit(event: FMLPostInitializationEvent) {

    }
}